### ISSUE TITLE
Fix S6931 FN: Route templates starting with `~/`

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#9002]
Route templates starting with `~/` seem to have an equivalent behavior to the ones starting with `/`: see [here](https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-8.0&source=recommendations#attribute-routing-for-rest-apis):

> In the preceding code, the Index method templates must prepend / or ~/ to the route templates. Route templates applied to an action that begin with / or ~/ don't get combined with route templates applied to the controller.

If so, the following is a False Negative:
```cs
public class A2Controller : Controller
{
    [Route("~/B")]
    public IActionResult B3([FromQuery] Person person) => Content("Hello");
}
```
since the following is a True Positive:
```cs
public class A2Controller : Controller // Noncompliant
{
    [Route("/B")]
    public IActionResult B3([FromQuery] Person person) => Content("Hello");
}
```

Alternative: We may also decide to consider `~/` as a more explicit way than `/`, for the user to say that the route should actually be attached to the web-application route, and not to the controller one. In that case, we should add `~/` as an exception for S6931.

Test case: https://sourcegraph.com/github.com/bitwarden/server@b9049cd699e8a80c585346f9c53caa40f38133f2/-/blob/src/Api/Controllers/InfoController.cs
